linera-service/src/node_service.rs

```diff
@@ -799,20 +801,65 @@ impl ChainStateViewExtension {
 }
 
 #[derive(MergedObject)]
-struct ChainStateExtendedView<C>(ChainStateViewExtension, Arc<ChainStateView<C>>)
+struct ChainStateExtendedView<C>(ChainStateViewExtension, ReadOnlyChainStateView<C>)
 where
     C: linera_views::common::Context + Clone + Send + Sync + 'static,
     ViewError: From<C::Error>,
     C::Extra: linera_execution::ExecutionRuntimeContext;
 
+/// A wrapper type that allows proxying GraphQL queries to a [`ChainStateView`] that's behind an
+/// [`OwnedRwLockReadGuard`].
+pub struct ReadOnlyChainStateView<C>(OwnedRwLockReadGuard<ChainStateView<C>>)
```

Comment:
Do we need to change any `async_graphql` cache settings to make sure this is freed as soon as the request is answered?

Reply:
See also my other comment

Reply:
I added a `#[graphql(cache_control(no_cache)))]` to `ChainStateView`. Hopefully that's enough :crossed_fingers: